File: py_aoc/1.py
from util import parse_file
from sets import Set

def parsefunc(s):
    return int(s)

seq = parse_file("../data/input_1", parsefunc)

# question 1
print("Question 1: %d" % sum(seq))

# question 2 - naive solution, but it works
previous_freqs = Set([0])
current_freq = 0
while 1:
    for delta in seq:
        current_freq += delta
        if current_freq in previous_freqs:
            print("Question 2: %d" % current_freq)
            quit()
        previous_freqs.add(current_freq)

File: go_aoc/1.go
// Make sure you have a file called input_1 with a number on each line in the
// same directory from which you are executing.
package main

import (
	"fmt"
	"os"
	"strconv"
)

func parsefunc(s []string) []int {
	buff := []int{}
	for _, c := range s {
		d, _ := strconv.Atoi(c)
		buff = append(buff, d)
	}
	return buff
}

func sum(r []int) int {
	retval := 0
	for _, i := range r {
		retval += i
	}
	return retval
}

func contains(s []int, e int) bool {
	for _, a := range s {
		if a == e {
			return true
		}
	}
	return false
}

func main() {
	current_freq := 0
	previous_freqs := []int{}
	freqs := parsefunc(readLines("../data/input_1"))
	// Question 1
	fmt.Printf("Question 1: %d\n", sum(freqs))
	// Question 2
	for true {
		for _, freq := range freqs {
			current_freq += freq
			if contains(previous_freqs, current_freq) {
				fmt.Printf("Question 2: %d\n", current_freq)
				os.Exit(0)
			}
			previous_freqs = append(previous_freqs, current_freq)
		}
	}
}
